middleware/probechecker/probechecker.go

```diff
@@ -0,0 +1,118 @@
+package probechecker
+
+import (
+	"fmt"
+
+	"github.com/gofiber/fiber/v2"
+)
+
+// ProbeChecker defines a function to check liveness or readiness of the application
+type ProbeChecker func(*fiber.Ctx) bool
+
+// ProbeCheckerHandler defines a function that returns a ProbeChecker
+type ProbeCheckerHandler func(ProbeChecker) fiber.Handler
+
+// Config is the config struct for the probechecker middleware
+type Config struct {
+	// Config for liveness probe of the container engine being used
+	//
+	// Optional. Default: func(c *Ctx) bool { return true }
+	IsLive ProbeChecker
+
+	// HTTP endpoint of the liveness probe
+	//
+	// Optional. Default: /livez
+	IsLiveEndpoint string
+
+	// Config for readiness probe of the container engine being used
+	//
+	// Optional. Default: nil
+	IsReady ProbeChecker
+
+	// HTTP endpoint of the readiness probe
+	//
+	// Optional. Default: /readyz
+	IsReadyEndpoint string
+}
+
+const (
+	DefaultLivenessEndpoint  = "/livez"
+	DefaultReadinessEndpoint = "/readyz"
+)
+
+func defaultLiveFunc(*fiber.Ctx) bool { return true }
+
+func probeCheckerHandler(checker ProbeChecker) fiber.Handler {
+	return func(c *fiber.Ctx) error {
+		if checker == nil {
+			return c.Next()
+		}
+
+		if checker(c) {
+			return c.SendStatus(fiber.StatusOK)
+		}
+
+		return c.SendStatus(fiber.StatusServiceUnavailable)
+	}
+}
+
+func checkRoute(path string, config *Config) string {
+	switch path {
+	case DefaultLivenessEndpoint, config.IsLiveEndpoint:
+		return "liveness"
+	case DefaultReadinessEndpoint, config.IsReadyEndpoint:
+		return "readiness"
+	default:
+		return ""
+	}
+}
+
+func New(config ...Config) fiber.Handler {
+	cfg := defaultConfig(config...)
+
+	checkers := map[string]fiber.Handler{
+		"liveness":  probeCheckerHandler(cfg.IsLive),
+		"readiness": probeCheckerHandler(cfg.IsReady),
+	}
+
+	return func(c *fiber.Ctx) error {
+		route := c.Route()
+		routeType := checkRoute(route.Path, &cfg)
+
+		if routeType != "" || route.Method != fiber.MethodGet {
+			handler, ok := checkers[routeType]
+
+			if !ok {
+				return fmt.Errorf("routeType of %s not found in checkers", routeType)
+			}
+
+			return handler(c)
+		}
+
+		return c.Next()
+	}
+}
```

Comment:
This all seems overly complicated. Why not do something similar to this:

```suggestion
func New(config ...Config) fiber.Handler {
	cfg := defaultConfig(config...)

	isLiveCheck := probeCheckerHandler(cfg.IsLive)
	isReadyCheck := probeCheckerHandler(cfg.IsReady)

	return func(c *fiber.Ctx) error {
		route := c.Route()
		
		if route.Method != fiber.MethodGet {
			return c.Next()
		}
		
		switch route.Path {
		case config.IsLiveEndpoint:
			return isLiveCheck(c)

		case config.IsReadyEndpoint:
			return isReadyCheck(c)
		}

		return c.Next()
	}
}
```